src/libs/elektra/kdb.c

```diff
@@ -999,6 +999,29 @@ KDB * kdbOpen (const KeySet * contract, Key * errorKey)
 		goto error;
 	}
 
+	// TODO (atmaxinger): improve
+	bool existingError = keyGetMeta (errorKey, "error") != NULL;
```

Comment:
IMO that's more of a nice-to-know. It's good if @atmaxinger knows that in future the recommended way to access meta keys is `ksLookupByName (keyMeta (errorKey), "meta:/error", 0)` (or possibly `keyGetMeta (errorKey, "meta:/error")`), but in the end it really doesn't matter if my `tree-surgeon` scripts have to replace one more call.

So maybe always use `meta:/` in future, but IMO there is really no need to change this now.

Reply:
@kodebach if your tool can do these changes automatically it is of course also fine for now. But at some point we need to learn writing correct code (not relying on legacy code being in place), otherwise we will introduce wrong code again...

Reply:
So we keep it for now and let the tool of @kodebach handle it.

Reply:
Yes, you can keep this. But for future PRs it's probably better to always use `meta:/` for meta keys. Just to get into the habbit and because there are already cases, where you need to `meta:/`.

Reply:
Do we have documentation for `keyMeta`? Because the API Doc still tells me that I should use `keySetMeta` and `keyGetMeta`.

Reply:
Beyond the API docs, I don't think `keyMeta` is documented yet. The docs are definitely outdated. `keyGetMeta`/`keySetMeta` might be removed (uncertain), but `keyMeta` definitely will be the main API for metadata in future, since you can already do everything necessary with it.

> Or is it just that I have to use the `meta:/` prefix for all keys I use with those functions?

When you use `keyMeta` then you definitely have to provide the `meta:/` namespace yourself. The fact that `keyGetMeta` and `keySetMeta` work without the `meta:/` prefix is a backwards compatibility thing. Nobody has yet wanted to update all uses, and my semi-automatic tool isn't ready yet.

The actual meta keys always use the `meta:/` prefix, because otherwise it wouldn't be a valid keyname. For example:

```c
printf ("%s\n", keyName (keyGetMeta (keyNew ("/foo", KEY_META, "type", "string", KEY_END), "type")));
// prints meta:/type
```

What you should do to get in the habit of things is write code with `meta:/` instead of without:

```c
Key * k = keyNew ("/foo", KEY_META, "meta:/type", "string", KEY_END);
keyGetMeta (k, "meta:/type");
keySetMeta (k, "meta:/type", "long");
```

This also makes it far less likely that you run into problems with one of the cases, where the `meta:/` is actually needed, e.g. things like:

```c
Key * k = keyNew ("/foo", KEY_META, "type", "string", KEY_END);

if (strcmp ("type", keyName (keyGetMeta (k, "type"))) == 0) {
   // doesn't work
}

if (strcmp ("meta:/type", keyName (keyGetMeta (k, "type"))) == 0) {
   // works
}
```